File: JavaScript/Hard.js
import Tyres from "./Tyres.js";

// Esta clase es para las llantas duras (Hard)
export default class Hard extends Tyres {
  // Constructor que delega en Tyres
  constructor(A, B, C, D, lapsMax, isWet, tyreColor) {
    super(A, B, C, D, lapsMax, isWet, tyreColor);
  }

  // Si supera la durabilidad máxima, entra a pits
  Durability(laps) {
    if (laps > this.durabilityMax) {
      this.Pits();
      return (
        `the tyres have been on track ${laps} laps\n` +
        `The Hards are being changed`
      );
    }
    return `the tyres have been on track ${laps} laps\nCan continue`;
  }

  // Presión recomendada para Hard
  PSI() {
    console.log("The air pressure required is 51");
  }
}

File: Go/Hard.go
package main
import "fmt"

type Hard struct{
	Tyres
}

func NewHard (A, B, C, D string, lapsMax int, isWet bool, tyreColor string) Hard{
	tyres := NewTyres(A, B, C, D, lapsMax, isWet, tyreColor)
	return Hard{Tyres:tyres}
}

func (h Hard) Durability(laps int) string{
	if (laps>h.durabilityMax){
		h.Pits()
		return fmt.Sprintf("The tyres have been on track %d laps\nThe Hards are being changed", laps)
	}else{
		return fmt.Sprintf("The tyre have been on track %d laps\nCan continue", laps)
	}
}

func (h Hard) PSI(){
	fmt.Println("The air pressure required is 51")
}